src/ros_filter.cpp

```diff
@@ -383,6 +397,120 @@ namespace RobotLocalization
   }
 
   template<typename T>
+  void RosFilter<T>::publishState(const ros::Time)
+  {
+      // Get latest state and publish it
+      nav_msgs::Odometry filteredPosition;
+
+      if (getFilteredOdometryMessage(filteredPosition))
+      {
+        worldBaseLinkTransMsg_.header.stamp = filteredPosition.header.stamp + tfTimeOffset_;
```

Comment:
It seems that most of this is only used if the first if condition is met; if not, it's overwritten, so IMO it should go inside the if clause.